addons/account/wizard/account_move_send.py

```diff
@@ -0,0 +1,464 @@
+        readonly=False,
+        store=True,
+    )
+
+    # == PRINT ==
+    enable_download = fields.Boolean(compute='_compute_enable_download')
+    checkbox_download = fields.Boolean(
+        string="Download",
+        compute='_compute_checkbox_download',
+        store=True,
+        readonly=False,
+    )
+
+    # == MAIL ==
+    enable_send_mail = fields.Boolean(compute='_compute_send_mail_extra_fields')
+    checkbox_send_mail = fields.Boolean(
+        string="Email",
+        compute='_compute_checkbox_send_mail',
+        store=True,
+        readonly=False,
+    )
+    display_mail_composer = fields.Boolean(compute='_compute_send_mail_extra_fields')
+    send_mail_warning_message = fields.Text(compute='_compute_send_mail_extra_fields')
+    send_mail_readonly = fields.Boolean(compute='_compute_send_mail_extra_fields')
+
+    mail_template_id = fields.Many2one(
+        comodel_name='mail.template',
+        string="Use template",
+        domain="[('model', '=', 'account.move')]",
+    )
+    mail_lang = fields.Char(
+        string="Lang",
+        compute='_compute_mail_lang',
+    )
+    mail_partner_ids = fields.Many2many(
+        comodel_name='res.partner',
+        string="Recipients",
+        domain=[('type', '!=', 'private')],
+        compute='_compute_mail_partner_ids',
+        store=True,
+        readonly=False,
+    )
+    mail_subject = fields.Char(
+        string="Subject",
+        compute='_compute_mail_subject',
+        store=True,
+        readonly=False,
+    )
+    mail_body = fields.Html(
+        string="Contents",
+        sanitize_style=True,
+        compute='_compute_mail_body',
+        store=True,
+        readonly=False,
+    )
+    mail_attachments_widget = fields.Json(
+        compute='_compute_mail_attachments_widget',
+        store=True,
+        readonly=False,
+    )
+
+    @api.model
+    def default_get(self, fields_list):
+        # EXTENDS 'base'
+        results = super().default_get(fields_list)
+
+        if 'move_ids' in fields_list and 'move_ids' not in results:
+            results['move_ids'] = [Command.set(self._context.get('active_ids', []))]
+
+        if 'move_ids' in results:
+            moves = self.env['account.move'].browse(results['move_ids'][0][2])
+            if len(moves.company_id) > 1:
+                raise UserError(_("You can only send from the same company."))
+            results['company_id'] = moves.company_id.id
+
+        return results
+
+    @api.model
+    def _get_mail_default_field_value_from_template(self, mail_template, lang, move, field, **kwargs):
+        return mail_template\
+            .with_context(lang=lang)\
+            ._render_field(field, move.ids, **kwargs)[move._origin.id]
+
+    @api.model
+    def _get_default_mode_from_moves(self, moves):
+        if all(x.is_invoice(include_receipts=True) for x in moves):
+            return 'invoice_single' if len(moves) == 1 else 'invoice_multi'
+        return None
```

Comment:
What else could it be?

Also, I don't see it being inherited, so I'd say keeping this in a separate function is useless and makes things harder to read